aten/src/ATen/native/TensorShape.cpp

```diff
@@ -3328,5 +3328,12 @@ at::Tensor diagonal_scatter(const at::Tensor& self, const at::Tensor& src, int64
     return output;
 }
 
+// The default implementation of lift is a no-op.
+// If TLS is set appropriately (for wrapper-tensor keys like Functionalize or functorch transforms),
+// then we'll dispatch to one of their implementations, which will properly lift the tensor into a wrapper.
+at::Tensor lift(const at::Tensor& self) {
```

Comment:
In which case can we actually reach this implementation?
Should this be an `TORCH_INTERNAL_ASSERT(false, ...)` ?

Reply:
This'll actually get called in all other cases from `tensor_new.cpp`. when we `return tensor.lift()`, it'll dispatch to `lift_functional()` if functionalization is enabled (or eventually, if the device is Lazy/XLA), and otherwise dispatch and and no-op.